src/js/components/sidenav.js

```diff
@@ -89,31 +89,43 @@ export default class Sidenav {
   }
   
   init() {
+    // Add click handler
+    this.element.addEventListener('click', this._handleClick, false);
+
+    // Expand all if openAllOnInit option is set
+    if (this.openAllOnInit) {
+      menuToggles.forEach(menuToggle => menuToggle.setAttribute('aria-expanded', 'true'));
+      childMenus.forEach(childMenu => childMenu.removeAttribute('hidden', ''));
+
+      return;
+    }
+
     // Get all the necessary DOM elements and convert to Arrays.
     const menuToggles = nodeListToArray(
       this.element.querySelectorAll(this.toggleSelector)
     );
     const childMenus = nodeListToArray(
       this.element.querySelectorAll(this.listSelector)
     );
-    
+
+    // Hide all child menus
+    childMenus.forEach(childMenu => childMenu.setAttribute('hidden', ''));
+
     menuToggles.forEach(menuToggle => {
```

Comment:
Something to think about here. You're calling `.forEach` four different times here. Given that you know a menu toggle and a child menus _should_ always have a set of matching data attribute values (`data-sidenav-toggle`/`data-sidenav-list`). You might not need to loop through the `childMenus` and `menuToggles` arrays multiple times each. You might think about how you can use some logic inside of of each `forEach` iteration. We can spend some time pairing on this if you want.

Reply:
This logic has been refactored. Have another look and let me know your thoughts!